card/src/main/java/com/adyen/checkout/card/StoredCardDelegate.kt

```diff
@@ -8,21 +8,42 @@
 
 package com.adyen.checkout.card
 
+import com.adyen.checkout.card.api.model.Brand
 import com.adyen.checkout.card.data.CardType
+import com.adyen.checkout.card.data.DetectedCardType
 import com.adyen.checkout.card.data.ExpiryDate
 import com.adyen.checkout.components.model.paymentmethods.StoredPaymentMethod
 import com.adyen.checkout.components.util.PaymentMethodTypes
 import com.adyen.checkout.components.validation.ValidatedField
 import com.adyen.checkout.core.log.LogUtil
 import com.adyen.checkout.core.log.Logger
+import kotlinx.coroutines.CoroutineScope
 
-private val NO_CVC_BRANDS: Set<CardType> = hashSetOf(CardType.BCMC)
+private val TAG = LogUtil.getTag()
 
+@Suppress("TooManyFunctions")
 class StoredCardDelegate(
     private val storedPaymentMethod: StoredPaymentMethod,
     cardConfiguration: CardConfiguration
 ) : CardDelegate(cardConfiguration) {
-    private val logTag = LogUtil.getTag()
+
+    private val cardType = CardType.getByBrandName(storedPaymentMethod.brand.orEmpty())
+    private val storedDetectedCardTypes = if (cardType != null) {
+        listOf(
+            DetectedCardType(
+                cardType,
+                isReliable = true,
+                showExpiryDate = true,
+                enableLuhnCheck = true,
+                cvcPolicy = when {
```

Comment:
Can this `cvcPolicy` logic be extracted to `CardDelegate` since it's also used in `NewCardDelegate. localDetectedCard`?

Reply:
No because they check different flags in the Configuration, we have `cardConfiguration.isHideCvc` and `cardConfiguration.isHideCvcStoredCard`